Archetypes guidance (#117)
ropensci-org/blog-guidance: authortechnical.Rmd

### Use any template with blogdown's New Post Addin {#blogdownaddin}

The blogdown New Post Rstudio addin creates the post draft in the correct location and fills the post YAML based on metadata you'll have entered.[^2]

* Install `whoami` and `blogdown` development version from GitHub

```r
whoami_path <- try(find.package("whoami"), silent = TRUE)

if (is(whoami_path, "try-error")) {
  install.packages("whoami")
}

remotes::install_github("rstudio/blogdown")
```

* Install Hugo if it wasn't already installed on your machine.

```r
blogdown::install_hugo()
```

* Re-start R.

* For your local website copy, after creating a branch, run blogdown's New Post addin from the RStudio addins menu or `blogdown:::new_post_addin()`

```{r blogdownaddin, echo = FALSE, fig.cap = "blogdown's New Post Addin."}
knitr::include_graphics("images/blogdownaddin.png", dpi = 25)
```

  * Enter a title, no need to worry about title case at this stage.
  * Enter your name if `whoami` wasn't able to guess it.
  * Choose the correct date.
  * Enter a new slug if the default one is too long.
  * Choose "blog" or "technotes" as a Subdirectory from the drop-down menu.

>   * Choose an Archetype, Rmd or md.

Reviewer: ```suggestion
  * Choose an Archetype, Rmd or md, from the drop-down menu.
```

## Start the post from a template {#templates}

> Prepare your post in your favorite editor in R Markdown **or** Markdown by starting with templates we provide in this book or using blogdown's handy New Post RStudio addin (recommended approach).

Author: I'd be strongly in favour of making this the recommended approach because as @steffilazerte noted, handling paths is a bit confusing. With the blogdown addin, there's more setup (installing two packages, installing Hugo) but then less mistakes possible.

knitr::include_graphics("images/hugocheckspassed.png")
```

[^1]: In Hugo speak, we'd say your post is a [leaf bundle](https://gohugo.io/content-management/page-bundles/#leaf-bundles).

> [^2]: If you don't use RStudio you can still use the addin, but the new post will be opened in [`getOption("editor")`](https://stat.ethz.ch/R-manual/R-devel/library/base/html/options.html).

Reviewer: I don't quite understand this "the new post will be opened in [`getOption("editor")`](https://stat.ethz.ch/R-manual/R-devel/library/base/html/options.html)"

Does the user need to set their preferred editor that way?

If yes, maybe rephrase

> [^2]: If you don't use RStudio you can still use the addin, but the new post will be opened in the editor returned by [`getOption("editor")`](https://stat.ethz.ch/R-manual/R-devel/library/base/html/options.html), that you might have configured or not.

Reviewer: ```suggestion
[^2]: If you don't use RStudio you can still use the addin, but the new post will be opened in the editor returned by [`getOption("editor")`](https://stat.ethz.ch/R-manual/R-devel/library/base/html/options.html), that you might need to configure.
```
